contrib/dracut/90zfs/zfs-load-key.sh.in

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# This script only gets executed on systemd systems, see mount-zfs.sh for non-systemd systems
+
+# There is a race between the zpool import and the pre-mount hooks, so we
+#   we wait for a pool to be imported
+while true; do
```

Comment:
This needs to kill itself after it sees that zpool-import-scan an zpool-import-cache failed, or it will just hang waiting for a pool on failed import

Reply:
and done (below)